### ISSUE TITLE
[rllib] Potential issue on SAC alpha loss

### ISSUE THREAD

**Issue by ethanluoyc** [ray-project/ray#15625]
### What is the problem?
*Ray version and other system information (Python version, TensorFlow version, OS):*
-NA-

I have noticed that the alpha loss used by rllib's SAC
https://github.com/ray-project/ray/blob/master/rllib/agents/sac/sac_torch_policy.py#L292
 to be different from other implementations. 
Here the `-torch.mean(model.log_alpha *
                                 (log_pis_t + model.target_entropy).detach())` is used but I have noticed that in other implementations it is usually 
`-torch.mean(torch.exp(model.log_alpha) *
                                 (log_pis_t + model.target_entropy).detach())`
For example, in softlearning

https://github.com/rail-berkeley/softlearning/blob/master/softlearning/algorithms/sac.py#L256

or pytorch-sac

https://github.com/poweic/pytorch_sac/blob/master/pytorch_sac/agent/sac.py#L127

log_alpha is used in 

https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/sac_continuous_action.py#L292. 

I would love to know the reason behind this discrepancy. Thanks a lot for the help!

I am not sure if a different loss function for alpha is intentional or a potential mistake

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".
- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by sven1977** [comment]
Hey @ethanluoyc . Great question! @hartikainen , one of the authors of the SAC paper actually implemented our SAC and I remember that it didn't really matter whether you are calculating the gradients for log(alpha) or alpha directly. It could be that the thought was that log adds some numeric stability here.

**Comment by ethanluoyc** [comment]
Thanks for the explanation! I suspect that's the reason as well.
